From e282a82a1f3145ff15dc2ba568c05ca068026834 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 9 Oct 2014 21:00:20 -0400
Subject: [fs] xfs: fix a couple error sequence jumps in xfs_mountfs()

Message-id: <1412888441-4181-27-git-send-email-sandeen@redhat.com>
Patchwork-id: 97324
O-Subject: [RHEL7.1 PATCH 26/47] [fs] xfs: fix a couple error sequence jumps in xfs_mountfs()
Bugzilla: 1145837
RH-Acked-by: Brian Foster <bfoster@redhat.com>
RH-Acked-by: Dave Chinner <dchinner@redhat.com>

Bugzilla: 1145837
Upstream Status: Committed upstream
Build Info: https://brewweb.devel.redhat.com/taskinfo?taskID=8082426
Tested: locally using xfstests

Thanks,
-Eric

    commit a70a4fa528faf6f22adce9a9067d1f4dfc332ade
    Author: Brian Foster <bfoster@redhat.com>
    Date:   Tue Jul 15 07:41:25 2014 +1000

    xfs: fix a couple error sequence jumps in xfs_mountfs()

    xfs_mountfs() has a couple failure conditions that do not jump to the
    correct labels. Specifically:

    - xfs_initialize_perag_data() failure does not deallocate the log even
      though it occurs after log initialization
    - xfs_mount_reset_sbqflags() failure returns the error directly rather
      than jump to the error sequence

    Signed-off-by: Brian Foster <bfoster@redhat.com>
    Reviewed-by: Dave Chinner <dchinner@redhat.com>
    Signed-off-by: Dave Chinner <david@fromorbit.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/xfs/xfs_mount.c b/fs/xfs/xfs_mount.c
index a9e29ea..f6255c0 100644
--- a/fs/xfs/xfs_mount.c
+++ b/fs/xfs/xfs_mount.c
@@ -850,7 +850,7 @@ xfs_mountfs(
       !mp->m_sb.sb_inprogress) {
   error = xfs_initialize_perag_data(mp, sbp->sb_agcount);
   if (error)
-   goto out_fail_wait;
+   goto out_log_dealloc;;
  }
 
  /*
@@ -922,7 +922,7 @@ xfs_mountfs(
    xfs_notice(mp, "resetting quota flags");
    error = xfs_mount_reset_sbqflags(mp);
    if (error)
-    return error;
+    goto out_rtunmount;
   }
  }
 
-- 
1.7.1